Sunrise/AppDelegate.swift

```diff
@@ -55,6 +61,36 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             AppRouting.showReservationWithId(reservationId)
         }
     }
+
+    func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
+        deviceToken = nil
+        saveDeviceTokenForCurrentCustomer()
+    }
+
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        let deviceToken = deviceToken.reduce("", {$0 + String(format: "%02X", $1)})
+        self.deviceToken = deviceToken
+        saveDeviceTokenForCurrentCustomer()
+    }
+    
+    func saveDeviceTokenForCurrentCustomer() {
+        if Commercetools.authState == .customerToken {
+            Customer.profile { result in
+                if let customerVersion = result.model?.version, result.isSuccess {
+                    var options = SetCustomTypeOptions()
+                    if let deviceToken = self.deviceToken {
+                        var type = ResourceIdentifier()
+                        type.id = "b1c3dbfb-68b9-407f-ba97-e1e0ef4c2bdb"
```

Comment:
I was also thinking about using creating a `Type` and conforming it to `ByKeyEndpoint` protocol, but unfortunately that doesn't work with mobile client scope.

I'm going to add an error log statement in case it fails, so at least the developer can see it in the console while running with device attached.